modules/services/dunst.nix

```diff
@@ -138,7 +146,7 @@ in {
           "status"
           "stock"
         ];
-      in concatStringsSep ":" (concatMap (theme:
+      in mkMerge (concatMap (theme:
```

Comment:
Why are you using `mkMerge` here?
`mkMerge` is only relevant to merge multiple, possibly conflicting module configuration.

While at it, can you refactor this to use variables? The current nesting makes it hard to understand what the function does.

Reply:
The idea is that the type of the option already encodes the implementation detail that it's a colon-separated list of paths, so the code can abstract that away. Arguably making the type something like "path or list of paths" with the appropriate merge semantics (like what NixOS does for [systemd unit options](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/system/boot/systemd-unit-options.nix#L18-L28)) would be more elegant, but it's more work.

I'll revert to `concatStringsSep ":"` if you still think `mkMerge` is confusing.

Reply:
> While at it, can you refactor this to use variables? The current nesting makes it hard to understand what the function does.

Is this better?

Reply:
Yup, very clean. I had no idea `cartesianProductOfSets` existed.